src/shapes/ActiveSelection.ts

```diff
@@ -86,6 +87,26 @@ export class ActiveSelection extends Group {
     }
   }
 
+  /**
+   * @override block ancestors/descendants of selected objects from being selected to prevent a circular object tree
+   */
+  canEnterGroup(object: FabricObject) {
+    if (
+      this.getObjects().some(
+        (o) => o.isDescendantOf(object) || object.isDescendantOf(o)
```

Comment:
Can change to the following as well, don't know what is better

```suggestion
        (o) => object.hasCommonAncestors(o)
```

Reply:
Better as is

Reply:
Now I understand this is not the expected behavior.
If an ancestor is trying to enter the descendant should exit